src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,46 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegisterException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_PASSWORD_AND_LOGIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+
+        if (user == null) {
+            throw new RegisterException("User can not be null");
+        }
+
+        for (User a : Storage.people) {
+            if (a.getLogin().equals(user.getLogin())) {
+                throw new RegisterException("Login already exists! Enter another value");
+            }
+        }
```

Comment:
The loop iterating over `Storage.people` assumes that `Storage.people` is a collection of `User` objects. Ensure that `Storage.people` is properly initialized and accessible. If `Storage.people` is not initialized, this will cause a `NullPointerException`.

Reply:
fixed